xds/src/main/java/io/grpc/xds/client/XdsClient.java

```diff
@@ -427,6 +435,25 @@ public interface ResourceStore {
     Collection<String> getSubscribedResources(ServerInfo serverInfo,
                                               XdsResourceType<? extends ResourceUpdate> type);
 
+    /**
+     * Like {@link #getSubscribedResources(ServerInfo, XdsResourceType)}, but limits the results to
+     * those matching the given authority.
+     */
+    @Nullable
+    Collection<String> getSubscribedResources(
+        ServerInfo serverInfo, XdsResourceType<? extends ResourceUpdate> type, String authority);
+
     Map<String, XdsResourceType<?>> getSubscribedResourceTypesWithTypeUrl();
+
+    /**
+     * Assigns the given resources to the given owner.  Generally, the owner is a ControlPlaneClient
+     * @param type the type of the resources
+     * @param resources the names of the resources
+     * @param owner the object that should take over ownership of the resources
+     */
+    default void assignResourcesToOwner(XdsResourceType<?> type, Collection<String> resources,
+                                        Object owner) {
+      // no-op - useful for test cases where everything is mocked
```

Comment:
nit:I'm confused by the comment.

Reply:
When everything is mocked there is no reason to do anything, so the cases where you want your test to implement ResourceStore you don't need to explicitly define this method that you don't want to use.